From 02359b064bf205f920c8320bd4c0ba62f8a36dcd Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Thu, 27 Oct 2022 14:24:09 +0530
Subject: [PATCH] ath12k: Fix the number of sounding dimensions above 80MHz
 issue

While copying HE PHY capabilities, The number of sound dimension above 80MHz
bits is reset with 0 masks instead of number of sound dimension below 80MHz
bits in the beacon template.

Fix it by avoiding the number of sound dimensions above 80MHz bits reset.

Fixes: 1e5aee4a23c63b (ath12k: propagation of EHT capabilities from
firmware to kernel)

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 08bcbbd..e675624 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -6371,7 +6371,7 @@ static void ath12k_mac_copy_he_cap(struct ath12k *ar,
 	he_cap_elem->phy_cap_info[0] &=
 		~IEEE80211_HE_PHY_CAP0_CHANNEL_WIDTH_SET_80PLUS80_MHZ_IN_5G;
 	he_cap_elem->phy_cap_info[5] &=
-		~IEEE80211_HE_PHY_CAP5_BEAMFORMEE_NUM_SND_DIM_ABOVE_80MHZ_MASK;
+		~IEEE80211_HE_PHY_CAP5_BEAMFORMEE_NUM_SND_DIM_UNDER_80MHZ_MASK;
 	he_cap_elem->phy_cap_info[5] |= ar->num_tx_chains - 1;
 
 	switch (iftype) {
-- 
2.17.1

